### ISSUE TITLE
Stickybox vertical/horizontal based on screen wide

### ISSUE THREAD

**Issue by humitos** [readthedocs/ethical-ad-client#194]
While working with stickybox I found that I'd like to have the vertical ad when it's floating in the bottom right (wide screen), but horizontal when it's injected at the bottom of the main content (narrow screen):

| Vertical | Horizontal |
| ----------- | --------------- |
| ![Screenshot_2024-05-21_10-16-07](https://github.com/readthedocs/ethical-ad-client/assets/244656/9a4aa5a6-3481-4649-b4bc-c0a7a1d3b0e1) | ![Screenshot_2024-05-21_10-16-25](https://github.com/readthedocs/ethical-ad-client/assets/244656/1f74f462-d736-4ebb-8632-174c7ea5c4fa) | 

I think it makes the ad to display nicer and more adapted to the content itself. Is this be a good idea? Would it be possible to communicate this to the ad client?

**Comment by ericholscher** [comment]
@davidfischer is probably better to answer this one.

**Comment by davidfischer** [comment]
The [stickybox](https://ethical-ad-client.readthedocs.io/en/latest/#stickybox) only floats at >1300px wide. Otherwise, it's supposed to collapse back to a static placement wherever you put the div (sidebar?). The stickybox can be used with the [`horizontal` class](https://ethical-ad-client.readthedocs.io/en/latest/#horizontal-image), but I don't think that's what you're doing here, right?

**Comment by humitos** [comment]
I'm not using the `horizontal` class by default because I don't want it to be horizontal all the time. I want to be `vertical` when the stickybox is floating on >1300px wide and `horizontal` in the other scenario. Is that possible? Does it make sense?

In the tests I've done, I found that it display nicer and that's why I opened this issue here.

**Comment by davidfischer** [comment]
The sticky box is *only* intended to be used at widths >1300px. On widths less than that, it collapses into a static placement.

**Comment by humitos** [comment]
Yes, my question is: when it collapses into a static placement, can it use the `horizontal` CSS class style?

**Comment by davidfischer** [comment]
> Yes, my question is: when it collapses into a static placement, can it use the `horizontal` CSS class style?

That isn't automatic and if you just set `horizontal` on it it would apply whether it is collapsed or not.

**Comment by humitos** [comment]
Yeah, that's what I realized while doing my tests. I'd like to change that behavior that's why I opened this issue here. Would that be a feature you accept in the client?

Another approach could be the ad client to trigger an event when it collapses. That way, I can subscribe to that event and add `horizontal` CSS class when it happens.

Does this make sense?

**Comment by davidfischer** [comment]
I don't like magically making something not horizontal (or making it horizontal) when there's a class that says the opposite. It seems not explicit.

**Comment by ericholscher** [comment]
Sounds like an event is perhaps the best option? But I think in general most folks aren't resizing windows, and we just need to default the right styles based on page width 🤔
